src/Components/Web/src/Forms/InputFile/SharedBrowserFileStream.cs

```diff
@@ -36,6 +35,8 @@ protected override async ValueTask<int> CopyFileDataIntoBuffer(long sourceOffset
                 SourceOffset = sourceOffset
             };
 
+            // We don't leverage the streaming mechanism directly as we want JavaScript to populate the 
+            // pre-allocated .NET memory.
```

Comment:
How important is this? Should we simplify this and just utilize streaming? This essentially works off of a `pull` model which breaks with the more `push` oriented way we do streaming elsewhere. This is why I avoided taking this approach with the `WebAssemblyJSDataStream.ReceiveData` to ensure implementation consistency & simplicity.

Reply:
Thanks for the comment about using the pre-allocated memory. That does explain some of the implementation.

TBH, given the very significant amount of extra complexity this brings, and the fact that streaming transfer is not something that happens commonly on the perf-critical render path, I would definitely be inclined to simplify it and use the existing binary interop with a pull-based pattern. I suspect the code would be *so* much simpler (also as mentioned at https://github.com/dotnet/aspnetcore/pull/33986/files#r668047758).

If, in the long term, we discover that people have a need to squeeze even more GC-efficiency out of streaming transfer, we can put in optimizations at that point. Even with a pull based model we could make it populate preallocated buffers without much code changes. But until we get to the point of needing that, I would expect streaming transfer to perform perfectly well on top of the binary interop mechanism.

Reply:
Thanks for the clarification, I'm inclined to agree in the interest of maintaining implementation simplicity. I've updated the `InputFile` implementation accordingly. It's much simpler now, doing away with `{Shared, Remote}BrowserFileStream` in favour of a standardized `BrowserFileStream` which simply using the underlying byte array interop mechanism. There are a couple of potential breaking changes, and I've left comments accordingly.